db/migrate/20201208093803_change_activity_programme_status_to_enum.rb

```diff
@@ -0,0 +1,25 @@
+class ChangeActivityProgrammeStatusToEnum < ActiveRecord::Migration[6.0]
+  def up
+    add_column :activities, :programme_status_integer, :integer
+
+    Activity.all.each do |activity|
+      programme_status = activity.programme_status.to_i
```

Comment:
On production, we have 131 activities where `programme_status` is `nil` and 5 where it's `Replace me`. Are there any risks with these all being converted to `0`..?

Reply:
Good shout. I've made this safer by using `Integer` with `exception: false`, so if it encounters anything that's not a number, it returns `nil`. I've also used `programme_status_before_type_cast`, rather than just `programme_status`, so we can migrate and revert safely with the enums in place.

Reply:
Didn't know about `Integer(value, exception: false)` 👍